### ISSUE TITLE
Tooltips for textfields

### ISSUE THREAD

**Issue by coco1979ka** [vaadin/vaadin-text-field#206]
Currently it seems not to be possible, to show tooltips for a textfield. I'm using paper-tooltip like this:
```
<vaadin-text-field label="Phone number" style="width:20em;" theme="small">
    <paper-tooltip>Insert your phone number here.</paper-tooltip>
</vaadin-text-field>
```
Are there plans to support this or is there a workaround? 
BTW: It works on the vaadin-checkbox.

I'm using 2.0.0-beta3

**Comment by tomivirkki** [comment]
You can make the tooltip work either by assigning it as the text-field's prefix:
```
<vaadin-text-field label="Phone number" style="width:20em;" theme="small">
  <paper-tooltip slot="prefix">Insert your phone number here.</paper-tooltip>
</vaadin-text-field>
```

or by using it externally:

```
<vaadin-text-field label="Phone number" style="width:20em;" theme="small" id="id_phone"></vaadin-text-field>
<paper-tooltip for="id_phone">Insert your phone number here.</paper-tooltip>
```

**Comment by jouni** [comment]
Thinking about this more generally, should we have a default `<slot>` for text-field, and the default content for that slot is the current `value` part (native `<input>`)? We kind of have that in `vaadin-dropdown-menu-text-field`, and I’m wondering if it that could/should be a feature?

**Comment by DiegoSanzVi** [comment]
What about using incubator-tooltip:
https://github.com/vaadin/incubator-tooltip
https://github.com/vaadin/incubator-tooltip-flow

It should work for every component, included vaadin-text-field.

![image](https://user-images.githubusercontent.com/40760222/48270020-a78f6580-e441-11e8-8e28-e301f1b1fe20.png)

**Comment by web-padawan** [comment]
> Thinking about this more generally, should we have a default <slot> for text-field,

Wrapping up the discussion, and for reference: this is being implemented in 2.3.0 but due to the complexity in brings, we do not promote it. For now we consider it rather a future improvement.

So, let me close this as answered per https://github.com/vaadin/vaadin-text-field/issues/206#issuecomment-373016090
